hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java

```diff
@@ -0,0 +1,121 @@
+
+    private final ConfigProviderImpl configProvider;
+
+    /**
+     * Creates a new instance of this class.
+     *
+     * @param configProvider the configuration provider
+     */
+    public SystemFileUpdateFacility(@NonNull final ConfigProviderImpl configProvider) {
+        this.configProvider = requireNonNull(configProvider, "configProvider must not be null");
+    }
+
+    /**
+     * Checks whether the given transaction body is a file update or file append of a special file and eventually
+     * notifies the registered facility.
+     *
+     * @param state the current state (the updated file content needs to be committed to the state)
+     * @param txBody the transaction body
+     */
+    public void handleTxBody(@NonNull final HederaState state, @NonNull final TransactionBody txBody) {
+        requireNonNull(state, "state must not be null");
+        requireNonNull(txBody, "txBody must not be null");
+
+        // Try to extract the file ID from the transaction body, if it is FileUpdate or FileAppend.
+        final FileID fileID;
+        if (txBody.hasFileUpdate()) {
+            fileID = txBody.fileUpdateOrThrow().fileIDOrThrow();
+        } else if (txBody.hasFileAppend()) {
+            fileID = txBody.fileAppendOrThrow().fileIDOrThrow();
+        } else {
+            return;
+        }
+
+        // Check if the file is a special file
+        final var configuration = configProvider.getConfiguration();
+        final var ledgerConfig = configuration.getConfigData(LedgerConfig.class);
+        final var fileNum = fileID.fileNum();
+        if (fileNum > ledgerConfig.numReservedSystemEntities()) {
+            return;
+        }
+
+        // If it is a special file, call the updater.
+        // We load the file only, if there is an updater for it.
+        final var config = configuration.getConfigData(FilesConfig.class);
+        try {
+            if (fileNum == config.addressBook()) {
+                logger.error("Update of address book not implemented");
+            } else if (fileNum == config.nodeDetails()) {
+                logger.error("Update of node details not implemented");
+            } else if (fileNum == config.feeSchedules()) {
+                logger.error("Update of fee schedules not implemented");
+            } else if (fileNum == config.exchangeRates()) {
+                logger.error("Update of exchange rates not implemented");
+            } else if (fileNum == config.networkProperties()) {
+                configProvider.update(getFileContent(state, fileID));
+            } else if (fileNum == config.hapiPermissions()) {
+                logger.error("Update of HAPI permissions not implemented");
+            } else if (fileNum == config.throttleDefinitions()) {
+                logger.error("Update of throttle definitions not implemented");
+            } else if (fileNum == config.upgradeFileNumber()) {
+
```

Comment:
what is supposed to go here? It would be good to have an issue number or at least a comment to describe future plans.

Reply:
The missing facilities (fees, throttles, ...) are in the process of being implemented. Part of their implementation is to add a call to their update functionality at this place. I believe it is a known requirement for the developers, but I will ensure that I make it explicit in the tickets.